conans/test/command/upload_test.py

```diff
@@ -186,6 +187,143 @@ def upload_unmodified_package_test(self):
         self.assertNotIn("Uploading conan_package.tgz", client2.out)
         self.assertIn("Package is up to date, upload skipped", client2.out)
 
+    def no_overwrite_argument_collision_test(self):
+        client = self._client()
+        client.save({"conanfile.py": conanfile,
+                     "hello.cpp": ""})
+        client.run("create . frodo/stable")
+
+        # Not valid values as arguments
+        error = client.run("upload Hello0/1.2.1@frodo/stable --no-overwrite kk", ignore_error=True)
+        self.assertTrue(error)
+        self.assertIn("ERROR", client.out)
+
+        # --force not valid with --no-overwrite
+        error = client.run("upload Hello0/1.2.1@frodo/stable --no-overwrite --force",
+                           ignore_error=True)
+        self.assertTrue(error)
+        self.assertIn("ERROR", client.out)
```

Comment:
Why don't check better the message? "'no_overwrite' argument cannot be used together with 'force'"

Reply:
just in case the message changes in the future... It is not prone change so I could add the whole message in this case

Reply:
If the message changes I will be glad to see a test failing, that's the purpose of a test.